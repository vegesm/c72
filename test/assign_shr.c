main() {
    extern iarr;
    int iarr[];
    int a,b;
    char c, d, carr[4];

    a = 5;
    printf("a >> 2 = %d\n", a =>> 2);
    printf("a = %d\n", a);

    a = 128;
    printf("a >> 3 = %d\n", a =>> 3);
    printf("a = %d\n", a);

    a = 24; b =3;
    printf("a >> b = %d\n", a =>> b);
    printf("a = %d\n", a);
    printf("b = %d\n\n", b);

    printf("iarr[0] >> 3 = %d\n", *iarr =>> 3);
    printf("iarr[0] = %d\n", iarr[0]);
    printf("iarr[1] = %d\n", iarr[1]);

    iarr[0] = 1;
    printf("iarr[1] >> 3 = %d\n", iarr[1] =>> 3);
    printf("iarr[0] = %d\n", iarr[0]);
    printf("iarr[1] = %d\n\n", iarr[1]);

    c = 5;
    printf("c >> 2 = %d\n", c =>> 2);
    printf("c = %d\n", c);
    c = -128;
    printf("c >> 3 = %d\n", c =>> 3);
    printf("c = %d\n", c);

    c = 5; d =2;
    printf("c >> d = %d\n", c =>> d);
    printf("c = %d\n", c);

    c = -128; d =2;
    printf("c >> d = %d\n", c =>> d);
    printf("c = %d\n\n", c);

    carr[0] = 5;
    carr[1] = -10;

    printf("carr[0] >> d = %d\n", *carr =>> d);
    printf("carr[0] = %d\n", carr[0]);
    printf("carr[1] = %d\n", carr[1]);

    carr[0] = 5;
    printf("carr[0] >> 2 = %d\n", *carr =>> 2);
    printf("carr[0] = %d\n", carr[0]);
    printf("carr[1] = %d\n", carr[1]);

    carr[0] = 5;
    printf("carr[1] >> d = %d\n", carr[1] =>> d);
    printf("carr[0] = %d\n", carr[0]);
    printf("carr[1] = %d\n", carr[1]);
}

iarr[] 1, 8, 10, 128;

