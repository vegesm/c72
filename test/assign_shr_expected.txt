a >> 2 = 1
a = 1
a >> 3 = 16
a = 16
a >> b = 3
a = 3
b = 3

iarr[0] >> 3 = 0
iarr[0] = 0
iarr[1] = 8
iarr[1] >> 3 = 1
iarr[0] = 1
iarr[1] = 1

c >> 2 = 1
c = 1
c >> 3 = -16
c = -16
c >> d = 1
c = 1
c >> d = -32
c = -32

carr[0] >> d = 1
carr[0] = 1
carr[1] = -10
carr[0] >> 2 = 1
carr[0] = 1
carr[1] = -10
carr[1] >> d = -3
carr[0] = 5
carr[1] = -3
